### PR TITLE
Implement Settings Page

### PR DESCRIPTION
**Pull request by AAAstorga**
I implemented the settings page for pollo. It looks like this:
image

Some notes:
- I did use Kotlin for this. Kotlin is very similar to Swift, and it really helped me onboard really quickly. If we are super against Swift, lmk and I will change it android.
- I also added a new way to manage secrets [https://medium.com/@geocohn/keeping-your-android-projects-secrets-secret-393b8855765d](url)
- I am looking for a lot of feedback because this is my first PR for android!! wooot

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -9,6 +12,8 @@ android {
         versionCode 1
         versionName "1.0"
         testInstrumentationRunner "android.support.test.runner.AndroidJUnitRunner"
+        buildConfigField("String", "web_client_id", keyProperties['web_client_id'])
```
Comment (young-k) [inline app/build.gradle]: this section should be alphabetized

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -1,33 +1,38 @@
...
+    <uses-permission android:name="android.permission.INTERNET"/>
 
     <application
-        android:allowBackup="true"
-        android:icon="@mipmap/ic_launcher"
-        android:label="@string/app_name"
-        android:roundIcon="@mipmap/ic_launcher_round"
-        android:supportsRtl="true"
-        android:theme="@style/AppTheme">
+            android:allowBackup="true"
```
Comment (young-k) [inline app/src/main/AndroidManifest.xml]: the indentation in this file should be 4, can you make sure your tabs are being converted to spaces properly?

File: app/src/main/java/com/cornellappdev/android/pollo/LoginActivity.java [hunk 3]
```diff
@@ -37,6 +37,16 @@ protected void onCreate(Bundle savedInstanceState) {
         mGoogleSignInClient = GoogleSignIn.getClient(this, gso);
     }
 
+    @Override
+    public void onBackPressed() {
+        // This is so when you are on the login activity and press the back button, you go to the home screen
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/LoginActivity.java]: this comment should probably be on the previous line, and shortened:
`// pressing the back button should go to home screen`

File: app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java [hunk 4]
```diff
@@ -106,6 +106,13 @@ protected void onStart() {
         }
     }
 
+    public void showSettings(View view){
+        Intent settings = new Intent(MainActivity.this, SettingsActivity.class);
+        startActivity(settings);
+    }
+
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/MainActivity.java]: too many new lines

File: app/src/main/java/com/cornellappdev/android/pollo/LoginActivity.java [hunk 5]
```diff
@@ -115,6 +125,7 @@ private void handleSignInResult(Task<GoogleSignInAccount> completedTask) {
             // Please refer to the GoogleSignInStatusCodes class reference for more information.
             //Log.w(TAG, "signInResult:failed code=" + e.getStatusCode());
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/LoginActivity.java]: can we get rid of this commented out code

File: app/src/main/java/com/cornellappdev/android/pollo/SettingsActivity.kt [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    fun logoutButtonClicked(view: View) {
+        val gso = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestProfile()
+                .build()
+        GoogleSignIn.getClient(this, gso).signOut();
+        finish();
+    }
+
```
Comment (young-k) [inline app/src/main/java/com/cornellappdev/android/pollo/SettingsActivity.kt]: delete?

File: app/src/main/res/layout/activity_main.xml [hunk 7]
```diff
@@ -46,6 +46,7 @@
                 android:textColor="@color/black"
                 android:textSize="30sp"
                 android:textStyle="bold"
+                android:onClick="showSettings"
```
Comment (young-k) [inline app/src/main/res/layout/activity_main.xml]: should probably be alphabetized

File: app/src/main/res/layout/activity_settings.xml [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+                      android:textSize="@dimen/header_font_size"
+                      android:layout_marginEnd="18dp"
+                      android:textStyle="bold"
+                      android:textColor="@color/black"/>
+
+            <TextView android:text="Unknown"
+                      android:layout_marginEnd="18dp"
+                      android:layout_marginTop="5.5dp"
+                      android:fontFamily="sans-serif-medium"
+                      android:textColor="#9EA7B3"
```
Comment (young-k) [inline app/src/main/res/layout/activity_settings.xml]: i don't think we should have any text colors here, this should all refer to colors that are defined in `colors.xml`

File: app/src/main/java/com/cornellappdev/android/pollo/LoginActivity.java [hunk 9]
```diff
@@ -115,6 +125,7 @@ private void handleSignInResult(Task<GoogleSignInAccount> completedTask) {
...
             //updateUI(null);
+            Log.d("Testing", e.getLocalizedMessage());
             Log.d("Testing", "Logged In Failed: " + e.getStatusCode());
```
Comment (2AStudios) [inline app/src/main/java/com/cornellappdev/android/pollo/LoginActivity.java]: Generally we try to leave Log messages out of our PR code. I'm not sure what the policy is while the app still isn't out, but it's not recommended to keep debugging tools in our production code.

File: app/src/main/java/com/cornellappdev/android/pollo/SettingsActivity.kt (hunk 6 above)
Comment (2AStudios) [reply app/src/main/java/com/cornellappdev/android/pollo/SettingsActivity.kt]: Yeah, I'd say that we shouldn't be mixing Java and Kotlin too often.

File: app/src/main/res/layout/activity_main.xml (hunk 7 above)
Comment (2AStudios) [reply app/src/main/res/layout/activity_main.xml]: This I feel is probably fine. I think it defaults to this in the visual editor.

File: app/src/main/res/layout/activity_settings.xml [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+            <TextView android:text="More"
+                      android:textSize="@dimen/header_font_size"
+                      android:textStyle="bold"
+                      android:layout_marginTop="20dp"
+                      android:layout_marginEnd="18dp"
+                      android:textColor="@color/black"/>
+
+            <TextView
+                    android:text="Let us know if you have any ideas, suggestions, or issues! Shake your phone to access the feedback form, or follow the link below."
+                    android:textColor="#9EA7B3"
```
Comment (2AStudios) [inline app/src/main/res/layout/activity_settings.xml]: General suggestion to use the "strings.xml" and "colors.xml" files a bit more often. It's a mild hassle compared to hardcoding it, but it is industry standard to move most things there.

**Pull request merged**